service/de_encoding/secretKeyCreator.go

```diff
@@ -2,42 +2,28 @@ package de_encoding
 
 import (
 	"crypto/sha256"
-	"fmt"
-	"io/ioutil"
-	"math/rand"
 )
 
-func GenerateRandomBytes(n int) ([]byte, error) {
-	b := make([]byte, n)
-	_, err := rand.Read(b)
-	if err != nil {
-		return nil, err
-	}
-	return b, nil
-}
+const keySize = 32 // length of secret key which is minimal to complete AES Encoding
+
+func PassToSecretKey(pass string) []byte {
 
-func shaEncrypt(fileNameFunc string) []byte {
 	h := sha256.New()
-	h.Write([]byte(fileNameFunc))
-	encryptedKeyWord := h.Sum(nil)
-	return encryptedKeyWord
-}
+	h.Write([]byte(pass))
 
-func CheckFile(name string) []byte {
-	randomKey, _ := GenerateRandomBytes(32)
-	fileName := shaEncrypt(name)
-	data, err := ioutil.ReadFile(string(fileName))
-	if err != nil {
-		fmt.Println(err)
+	encryptedKeyWord := h.Sum(nil)
+	if h.Size() < 32 {
+		dif := 32 - h.Size()
+		return CreateArray(dif, encryptedKeyWord)
 	}
-	if data != nil {
-		return data
-	} else {
-		err := ioutil.WriteFile(string(fileName), randomKey, 0777)
+	return encryptedKeyWord[:keySize]
 
-		if err != nil {
-			fmt.Println(err)
-		}
-		return randomKey
+}
+func CreateArray(diff int, arr []byte) []byte {
```

Comment:
This looks like a tricky function to grasp. Why is it public? I could suggest instead of filling in with the same number (why 65? :) ) you could try and copy already filled part of array into empty part. It could be helpful to preallocate a slice with required size.